gradio/components/image_editor.py

```diff
@@ -95,8 +120,10 @@ class ImageEditor(Component):
     EVENTS = [
         Events.clear,
         Events.change,
+        Events.input,
         Events.select,
         Events.upload,
+        Events.apply,
     ]
     data_model = EditorData
 
```

Comment:
We should add a `format` param instead of hardcoding to webp:
```
        format: str = "webp",
```

Reply:
I don't know that a format param really makes sense here, the images are all either generated by or used by the image editor internally.

The only time this would ever have any impact is if a user is using the `filepath` type. In all other cases the images are either utilised by the imageeditor frontend internally or converted to np or PIL images before the users sees them.

It will also potentially undo the added performance that webp brings. I personally think we should require this to be webp. In the rare cases where a user definitely needs an image file and that files needs to be something other than webp, it is easier for them to convert it.

Reply:
@pngwn there are two cases I see where a user might want to use the `format` parameter:

(1) as you said, if the `ImageEditor` component is being used as an input component with `type="filepath"`. 

(2) If the `ImageEditor` is being used as an output component. Currently, the `ImageEditor` as an output component is exactly the same as the `Image` component, but in the future, we might want to expose the different layers as well. In this case, users might want to be able to set the format as well.

In both cases, while it is true that other formats are less performant, we should set the default to `webp` and then let users change it if needed -- this would be consistent with other components, such as `Image` or `Video`

Reply:
That makes sense. Will add the kwarg.